wrappers/services.go

```diff
@@ -274,6 +290,10 @@ func StopServices(apps []*snap.AppInfo, reason snap.ServiceStopReason, inter int
 		if !app.IsService() || !osutil.FileExists(app.ServiceFile()) {
 			continue
 		}
+		// We can't stop services that run under a user mode systemd
+		if app.IsUserService() {
```

Comment:
The fact that we don't stop services is different than for system services. Can this break user services?

Reply:
It's more the fact that snapd can't talk to the user mode systemd instance (or multiple instances if you happen to have a multi-seat system).

As for what could break, it'd be the same things that can break for snap applications running over an upgrade or uninstall.

Reply:
This is problematic. We either need to find a way to stop them or we need infrastructure enough to inform the snap that is about to be updated and for it to do the right thing with those?  

@jhenstridge also does this mean that "snap start/stop" etc don't work for these services?

Reply:
Correct: "snap start/stop" will not do anything because snapd can not talk to the user session systemd instance.

So we're essentially in the same position as existing long running snap applications (e.g. a messaging app started via xdg autostart that runs minimised and is only visible through the notifications it posts).

We are in a better position than the status quo though, in that if snapd grows a user session agent to handle upgrades, there is a defined method of restarting these daemons (unlike a long running GUI application).